packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1199,3 +1298,29 @@ class FollowerLayer extends ContainerLayer {
     properties.add(new TransformProperty('transform', getLastTransform(), defaultValue: null));
   }
 }
+
+/// A composited layer which annotates it's children with a value.
+class AnnotatedRegionLayer<T> extends ContainerLayer {
+
+  /// Creates a new annotated layer.
+  AnnotatedRegionLayer(this.value);
+
+  /// The value which annotates the children.
+  final T value;
```

Comment:
before landing, please take a second pass at these docs bearing in mind the writing prompts at https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-useless-documentation